src/swupd/manifest_test.go

```diff
@@ -254,3 +258,90 @@ func TestReadManifestFromFileEmpty(t *testing.T) {
 		t.Error("ReadManifestFromFile did not raise error on empty file")
 	}
 }
+
+func compareFiles(file1path string, file2path string) (bool, error) {
+	chunkSize := 64000
```

Comment:
Before you start reading, do a couple of os.Stat calls. If the sizes differ then you don't need to do all the I/O to read them.

64000 is a reasonable number, but might be worth switching to 65536 in the vague hope that you might get some zero copying advantages from the kernel